rosa_tutorials/rosa-mobb-external-dns-for-custom-domain-tutorial.adoc

```diff
@@ -0,0 +1,328 @@
+:_content-type: ASSEMBLY
+[id="rosa-mobb-external-dns-for-custom-domain-tutorial"]
+= Tutorial: Deploying the external DNS Operator with a custom domain in ROSA
+include::_attributes/attributes-openshift-dedicated.adoc[]
+:context: ROSA-mobb-external-dns-for-custom-domain-tutorial
+
+toc::[]
+
+//Mobb content metadata
+//Brought into ROSA product docs 2023-09-20
+//---
+//date: '2021-06-10'
+//title: External DNS for ROSA custom domain
+//weight: 1
+//tags: ["AWS", "ROSA"]
+//authors:
+//  - Chris Kang
+//  - Dustin Scott
+//---
+
+Configuring the `Custom Domain` Operator requires a wildcard CNAME DNS record in your Route53 hosted zone. If you do not want to use a wildcard record, you can use the `External DNS` Operator to create individual entries for routes.
+
+This document will guide you through deploying and configuring the External DNS Operator with a custom domain in Red Hat OpenShift Service on AWS (ROSA).
+
+[IMPORTANT]
+====
+The `External DNS` Operator does not support STS yet and uses long lived IAM credentials. This guide will be updated once STS is supported.
+====
+
+[id="rosa-mobb-external-dns-for-custom-domain-prerequisites"]
+== Prerequisites
+
+* A user account with `dedicated-admin` privileges
+* Access to a ROSA cluster with the latest version of the `oc` CLI installed
+* The latest AWS CLI
+* The latest `rosa` CLI
+* Route53 hosted zone
+* A domain
+
+[id="rosa-mobb-external-dns-for-custom-domain-deploy"]
+== Deploy
+
+Deploy the External DNS Operator by completing the following sections.
+
+=== Setup environment
+
+. Set your email and your domain:
++
+[source,terminal]
+----
+$ export EMAIL=<YOUR-EMAIL>
+----
++
+[source,terminal]
+----
+$ export DOMAIN=<YOUR-DOMAIN>
+----
++
+. Set the remaining environment variables:
++
+[source,terminal]
+----
+$ export SCRATCH_DIR=/tmp/scratch
+$ export ZONE_ID=$(aws route53 list-hosted-zones-by-name --output json \
+--dns-name "$DOMAIN." --query 'HostedZones[0]'.Id --out text | sed 's/\/hostedzone\///')
+$ mkdir -p $SCRATCH_DIR
+----
+
+=== Custom domain
+////
+To be added in.
+[NOTE]
+====
+For information on deploying the External DNS Operator without using a wildcard certificate, see ../rosa_tutorials/rosa-mobb-dynamic-certificate-custom-domain[Dynamically issuing certificates with a custom domain in ROSA].
+====
+////
+. Create a TLS key pair for the custom domain using certbot:
++
+[NOTE]
+====
+Skip this step if you already have a key pair.
+====
++
+[source,terminal]
+----
+$ certbot certonly --manual \
+  --preferred-challenges=dns \
+  --email $EMAIL \
+  --server https://acme-v02.api.letsencrypt.org/directory \
+  --agree-tos \
+  --config-dir "$SCRATCH_DIR/config" \
+  --work-dir "$SCRATCH_DIR/work" \
+  --logs-dir "$SCRATCH_DIR/logs" \
+  -d "*.$DOMAIN"
+----
++
+. Create a new TLS secret for the custom domain:
++
```

Comment:
You should probably combine these commands into one Terminal example. Just a little confusing structured like this without comments separating the examples.